src/renderer/views/pool/PoolChartView.tsx

```diff
@@ -0,0 +1,119 @@
+import React from 'react'
+
+import * as RD from '@devexperts/remote-data-ts'
+import BigNumber from 'bignumber.js'
+import * as FP from 'fp-ts/function'
+import * as O from 'fp-ts/Option'
+import { useObservableState } from 'observable-hooks'
+
+import { PoolChart } from '../../components/pool/PoolChart'
+import { useMidgardContext } from '../../contexts/MidgardContext'
+import { SelectedPricePoolAsset } from '../../services/midgard/types'
+import { GetDepthHistoryIntervalEnum, GetSwapHistoryIntervalEnum } from '../../types/generated/midgard'
+import { getEoDTime, getWeekAgoTime } from './PoolChartView.helper'
+
+type Props = {
+  isLoading?: boolean
+  priceRatio: BigNumber
+}
+
+export const PoolChartView: React.FC<Props> = ({ isLoading, priceRatio }) => {
+  const {
+    service: {
+      pools: { selectedPricePoolAsset$, getSwapHistory$, getDepthHistory$ }
+    }
+  } = useMidgardContext()
+
+  const selectedPricePoolAsset = useObservableState<SelectedPricePoolAsset>(selectedPricePoolAsset$, O.none)
+
+  const curTime = getEoDTime()
+  const weekAgoTime = getWeekAgoTime()
+
+  const [swapAllHistoryRD] = useObservableState(
```

Comment:
Currently data for `swapAllHistoryRD` (all data) and `swapWeekHistoryRD` (weekly data) is loaded with two request. Since an user can switch between `all` and `weekly` data,  but maybe he will never switch), my suggestion is to load data only we really show in the chart (weekly OR all data). But that's more an optimization thing we don't need atm.

Reply:
Fixed by #1261